### DISCUSSION TITLE
Disable "Move already placed notes when changing the offset/BPM" by default in the editor

### DISCUSSION THREAD

**Discussion by trollocat** [ppy/osu#17060] [Ideas] [answered]
_By default, this option is enabled when you get into the editor. Even when you disable it, if you exit the map and enter again it will re enable itself._
This is the **most map-breaking feature ever**, no one really uses it and when you have 2 or more timing points on your map and make any timing change with it enabled, it repositions ALL the notes (affected by any timing point) and it always does so incorrectly, because it assumes you have single bpm across all the map, which is almost **never the case**. The worst thing is **you can't undo this**, so it basically breaks the entirety of your map without being able to do anything about it, in the blink of an eye you're back to square one... just because you didn't see a small pink radio button that you never use.

Every fellow mapper I know has gotten their maps **obliterated** multiple times by this feature alone, we always curse it and yet it's still here after so many years. It's really unbelievable.

The best and almost mandatory solution is to **disable this feature by default** and make it not re enable itself every time you open the editor. This would be such a relief for every mapper out there. I swear having so many maps undone because of this over the years demotivated me from mapping complexly timed songs, so I really hope this can change for the better of all the mapping community. _Thanks._

**Comment by peppy** [comment, marked as answer]
I guess you're proposing this is changed in stable?

This is all very hyperbolised and any change is easy to fix (using ctrl-a) FWIW.

**Reply by trollocat** [reply]
Yes, to be changed on stable. It's not fixable by Ctrl+A and doing something to all notes, because the feature moves the notes' offset independently at random snappings, so there's no way of moving all the notes to the previous state because they're all shuffled, the only way to fix it is to manually snap again circle by circle.

Maybe I didn't specify that by "reposition" I meant that this feature I'm talking about **resnaps** notes on the timeline, as in moving the offset of the notes around, not the actual (X,Y) position. My bad.

**Reply by trollocat** [reply]
![image](https://user-images.githubusercontent.com/68620390/156485193-35be7938-3543-4b9a-aa18-1985800d6bf7.png)
I mean this is what it looks like after it does its "magic"... how can you fix this without Ctrl+Z? No way, you just scrap everything and start over

**Reply by peppy** [reply]
Ctrl-L works as well. Or you can use the timing panel instead, which is set with better defaults for advanced scenarios.

The one on the timing screen that you propose as an issue is intended for simple scenarios where it works 99% of the time, and gives the user what they expect (up to 1/8 snaps). With it turned off, the beatmap would be just as broken (the user would have to move each object one by one).

Note that this is a stable-only issue but I can't transfer discussion threads so I'll just leave this response here.

**Reply by trollocat** [reply]
> Ctrl-L works as well.

Yes, but only if you're quick to realize and didn't save at any time after the fact (which any mapper will fail automatically because you basically auto Ctrl+S every time you do something). But still this shouldn't be a solution for anything really, returning to last saved state doesn't solve anything in the big picture, this shouldn't be happening in the first place.


> Or you can use the timing panel instead, which is set with better defaults for advanced scenarios.

Nothing helps here, if I try to use any of these functions, they won't work correctly because notes are so misplaced that it's impossible that any of them will get it right like it was before the disaster happens. I use them frequently and know exactly how they work (for example 1/24 to snap multiple snappings).
![image](https://user-images.githubusercontent.com/68620390/156494631-181dc858-304b-45c4-b57e-301ad952d21a.png)


> The one on the timing screen that you propose as an issue is intended for simple scenarios where it works 99% of the time, and gives the user what they expect (up to 1/8 snaps).

I get what you say, but this option is unnecessary, as those mappers could simply move the notes along with Ctrl+A (everyone knows how to do this) or use the "Move all notes in time..." menu, completely negating the purpose of having this 'helper' function enabled by default.


> With it turned off, the beatmap would be just as broken (the user would have to move each object one by one).

Hard disagree. Having the option disabled will NEVER break the map for these cases, because they can always Ctrl+Z something if they screw up (unlike the other way around I am discussing which can't be reverted unless by lucky one-time quick-reflex Ctrl+L that never happens because of the workflow of mapping).


Also if they really wanted to use this function -in the theoretical case in which it's not enabled by default- they can enable it just when they need to. And it makes sense, you only want to use such a high-impact tool on your map only when you need to, why is it turned on by default? Just because "most" of the users may find it useful? I mean the seconds you saved for them aren't worth the hours of wasted work for the people who didn't want to have this thing enabled without their consent. Get what I mean? It's not convenient to leave it enabled because it is destructive, while if it's disabled it isn't destructive (user can just Ctrl+Z their own mistakes).

I also should add that entering Autoplay and returning to editor also re-enables the feature after you disable it.

For example: I am currently mapping a song that has +100 timing points and I really like testing the map between changes. How many times in a single session did I get my notes completely randomized by this function? Four. Four remaps of the entire thing. And I am fully aware of all the ways to prevent it, it's just too much effort to constantly disable the option every time I do something so I end up forgetting about it.

You can ask anyone who mapped complex timing maps and they will all have really bad experiences with this function. It's destructive without reason. I hope you understand this issue more clearly now, when you balance things out, in the end everyone will be happier with this function disabled by default (and not reactivating itself at any given point unless by user input). Please don't leave this unresolved as I think it's a huge drawback of the stable editor that could greatly improve the usage for experienced mappers.

**Reply by peppy** [reply]
Ctrl+A and move will not work for BPM changes. The provided setting *does* work.

Applying offset changes via the timing panel (f6) with default settings will do what you want.

I would recommend you always use the f6 panel as it provides the options you require.

**Reply by peppy** [reply]
I'm going to lock this discussion as it's in the wrong place and also a touch misguided. I may look at changing the way the checkbox works (ie. not having it snap but only perform movement), but do note that the reason it's not done this way is it will cause integer rounding issues when multiple applications are made.

If you would like to discuss further, please open a forum thread or use https://github.com/ppy/osu-stable-issues/issues, but also make sure you have a good understanding of what I've said. Because from your last message it's clear that you aren't fully grasping the picture.

**Reply by peppy** [reply]
@trollocat i've changed the behaviour as proposed in the latest stable release, so it no longer snaps. we'll see how people react.
